### PR TITLE
User notes

### PR DESCRIPTION
**Pull request by gardnerjack**
New table `UserNote` that stores `aaid`, `note`, and `setByAaid`. This maintains a one-to-one mapping of `aaid` to a note, so when a note is created a new row is created, when the note is edited the existing row is updated, and when the note is removed the row is deleted.

When fetching the current state of the runner, notes are attached to the list of user objects if the requesting user is an admin. Notes and the associated controls are also only displayed to admins.

### CODE REVIEW

File: src/lib/PermissionService.ts [hunk 1]
```diff
@@ -36,6 +36,37 @@ class PermissionService {
...
+        where: {
+          aaid,
+        },
+      },
+    ))[0] > 0 || // Check if a row was updated, otherwise create a new row
+      (await UserNote.create<UserNote>({
+        aaid,
+        note,
+        setByAaid: setter.aaid,
+      }));
```
Comment (lukebatchelor) [inline src/lib/PermissionService.ts]: This feels a little messy and hard to read. Could we do something like:

```
const numNotesForUser = await UserNote.count({ where: { aaid }});
if (numNotesForUser === ) { // create the first note
  await UserNote.create<UserNote>({ aaid, note, setByAaid: setter.aaid });
} else { // update the existing one
  await UserNote.update({ note, setByAaid: setter.aaid }, { where: aaid });
}
```

File: src/lib/PermissionService.ts [hunk 2]
```diff
@@ -36,6 +36,37 @@ class PermissionService {
...
+  };
+
+  removeUserNote = async (aaid: string): Promise<void> => {
```
Comment (lukebatchelor) [inline src/lib/PermissionService.ts]: Absolutely not worried, but do you think we actually need to destroy them? Or just mark them blank?
Definitely not worth changing, just wondering if there was a reason for it

File: src/static/current-state/components/tabs/UsersList.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
```
Comment (lukebatchelor) [inline src/static/current-state/components/tabs/UsersList.tsx]: This is super clean, nice!

File: src/lib/PermissionService.ts (hunk 2 above)
Comment (gardnerjack) [reply src/lib/PermissionService.ts]: Just in the interest of maintaining a specific pattern for this table. Where the row exists if the note exists, and there is one row/note per user.

**Pull request merged**